reconcile/vpc_peerings_validator.py

```diff
@@ -14,10 +16,162 @@
+                        comparing_vpc_dict, mesh_vpc_dict = cidr_mesh_finder(
+                            aws_account_uid, tags_key, tags_value, awsapi
+                        )
+                        vpc_peering_info = {
+                            "provider": peering.provider,
+                            "vpc_peering": {
+                                "comparing_vpc_dict": comparing_vpc_dict,
+                                "mesh_vpc_dict": mesh_vpc_dict,
+                            },
+                        }
+                        if cluster.name not in peerings_enteries_dict:
+                            peerings_enteries_dict[cluster.name] = {
+                                "cidr_block": cluster.network.vpc,  # type: ignore[union-attr]
+                                "providers": [],
+                            }
+                        peerings_enteries_dict[cluster.name]["providers"].append(
+                            vpc_peering_info
+                        )
+                if peering.provider == "account-vpc":
+                    cidr_block = str(peering.vpc.cidr_block)  # type: ignore[union-attr]
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_name": peering.vpc.name,  # type: ignore[union-attr]
+                        "region": peering.vpc.region,  # type: ignore[union-attr]
+                        "cidr_block": cidr_block,
+                    }
+                    if cluster.name not in peerings_enteries_dict:
+                        peerings_enteries_dict[cluster.name] = {
+                            "cidr_block": cluster.network.vpc,  # type: ignore[union-attr]
+                            "providers": [],
+                        }
+                    peerings_enteries_dict[cluster.name]["providers"].append(
+                        vpc_peering_info
+                    )
+                if peering.provider in (
+                    "cluster-vpc-requester",
+                    "cluster-vpc-accepter",
+                ):
+                    vpc_peering_info = {
+                        "provider": peering.provider,
+                        "vpc_peering": {
+                            "vpc_name": peering.cluster.name,  # type: ignore[union-attr]
+                            "cidr_block": peering.cluster.network.vpc,  # type: ignore[union-attr]
+                        },
+                    }
+                    if cluster.name not in peerings_enteries_dict:
+                        peerings_enteries_dict[cluster.name] = {
+                            "cidr_block": cluster.network.vpc,  # type: ignore[union-attr]
+                            "providers": [],
+                        }
+                    peerings_enteries_dict[cluster.name]["providers"].append(
+                        vpc_peering_info
+                    )
+
+    overlaps_peering_entries_dict = find_cidr_duplicates_and_overlap(
```

Comment:
We don't need to check overlaps after getting all clusters' info. instead, we can check per cluster. 
so I suggest changing following
```
peerings_enteries_dict = {} 
for cluster in clusters:
     ...append info to peerings_enteries_dict
find_cidr_duplicates_and_overlap(peerings_enteries_dict)
```
to
```
for cluster in clusters:
     peerings_enteries_dict = {} 
     ...append info to peerings_enteries_dict
     find_cidr_duplicates_and_overlap(peerings_enteries_dict)
```
peerings_enteries_dict only stores the vpc cidr for one cluster, Once one cluster fails the check, we can break and make the whole fail.